chain/client/src/sync.rs

```diff
@@ -469,12 +472,13 @@ impl BlockSync {
             return Ok(true);
         }
 
-        let reference_hash = match &self.last_request {
-            Some(request) if chain.is_chunk_orphan(&request.hash) => request.hash,
-            _ => chain.head()?.last_block_hash,
-        };
-
+        // reference_hash is the last block on the canonical chain that is in store (processed)
```

Comment:
can you explain a little bit more about the 'last_request' path in this match? when does it happen?

Reply:
honestly I think this case could be removed. When I digged to the original PR, it said "download either the block that builds directly on top of head, or one block ahead if the next block is already in the blocks with missing chunks.", which made sense. The last_request path was referring to the case "if the next block is already in the blocks with missing chunks". But then there is another PR later to fix a bug " if the current head is not on the canonical header chain, the block sync would continuously try to fetch a block that build on top of the current head, which is not what needs to be fetched". The fix added lines https://github.com/near/nearcore/blob/master/chain/client/src/sync.rs#L499-L516 to deal with the fork case, but now these lines made the "next block is already in the blocks with missing chunks" case obsolete because `reference_hash` now is the effective head in all cases.

I'll remove the case and add some comments in the code.